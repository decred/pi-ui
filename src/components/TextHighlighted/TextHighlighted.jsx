import PropTypes from "prop-types";
import React from "react";
import Text from "../Typography/Text/Text.jsx";
import { classNames, idPropTypeCheckForTruncatedComponents } from "../../utils";
import styles from "./styles.module.css";

const TextHighlighted = ({ id, truncate, className, style, children }) => (
  <Text
    {...{
      id,
      truncate,
      style,
      className: classNames(styles.contentWrapper, className),
    }}
    style={style}>
    {children}
  </Text>
);

TextHighlighted.propTypes = {
  id: idPropTypeCheckForTruncatedComponents,
  truncate: PropTypes.bool,
  className: PropTypes.string,
  style: PropTypes.object,
  children: PropTypes.node,
};

TextHighlighted.defaultProps = {
  truncate: true,
};

export default TextHighlighted;
