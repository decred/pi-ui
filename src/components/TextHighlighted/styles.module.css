.contentWrapper {
  padding: 0.3rem 1rem;
  word-break: break-all;
  background-color: var(--copyable-text-background-color);
  border-radius: 0.4rem;
}
