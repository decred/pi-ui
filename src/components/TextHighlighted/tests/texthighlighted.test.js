import React from "react";
import TextHighlighted from "../TextHighlighted";
import { create } from "react-test-renderer";
import {
  defaultLightTheme,
  ThemeProvider,
  DEFAULT_LIGHT_THEME_NAME,
} from "../../../theme";

describe("TextHighlighted component", () => {
  test("Matches snapshot", () => {
    const texthighlighted = create(
      <ThemeProvider
        themes={{ [DEFAULT_LIGHT_THEME_NAME]: defaultLightTheme }}
        defaultThemeName={DEFAULT_LIGHT_THEME_NAME}>
        <TextHighlighted id="text-id">test</TextHighlighted>
      </ThemeProvider>
    );
    expect(texthighlighted.toJSON()).toMatchSnapshot();
  });
});
