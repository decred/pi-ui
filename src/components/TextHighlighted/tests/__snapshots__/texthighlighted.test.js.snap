// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextHighlighted component Matches snapshot 1`] = `
<span
  className="text textAlignStart weightRegular sizeNormal colorDefault contentWrapper"
  id="text-id"
>
  test
</span>
`;
